<launch>

  <!-- camera configurations -->
    <!-- mode: topic or service -->
    <arg name="mode" default="topic"/> 
    <!-- rgb topic -->
    <arg name="rgb" default="/camera/color/image_raw"/>
    <!-- aligned depth to rgb topic -->
    <arg name="depth" default="/camera/aligned_depth_to_color/image_raw"/>
    <!-- camera info topic -->
    <arg name="camera_info" default="/camera/color/camera_info"/>

  <!-- UOAIS-Net configurations -->
    <arg name="config_file" default="configs/R50_rgbdconcat_mlc_occatmask_hom_concat.yaml" />
    <arg name="confidence_threshold" default="0.5"/>
    
  <!-- Foreground segmentation  -->
    <!-- CG-Net  -->
    <arg name="use_cgnet" default="false"/>
    <arg name="cgnet_weight" default="foreground_segmentation/rgbd_fg.pth"/>
    <!-- RANSAC plane segmentation -->
    <arg name="use_planeseg" default="false"/>
    <arg name="ransac_threshold" default="0.003"/>
    <arg name="ransac_n" default="3"/>
    <arg name="ransac_iter" default="10"/>

  <!-- UOAIS-Net node  -->
  <node name="uoais"  pkg="uoais" type="uoais_node.py" output="screen" >
    	<param name="mode" value="$(arg mode)" />
      <param name="rgb" value="$(arg rgb)" />
      <param name="depth" value="$(arg depth)" />
      <param name="camera_info" value="$(arg camera_info)" />
      <param name="config_file" value="$(arg config_file)" />
      <param name="confidence_threshold" value="$(arg confidence_threshold)" />
      <param name="use_cgnet" value="$(arg use_cgnet)" />
      <param name="cgnet_weight" value="$(arg cgnet_weight)" />
      <param name="use_planeseg" value="$(arg use_planeseg)" />
      <param name="ransac_threshold" value="$(arg ransac_threshold)" />
      <param name="ransac_n" value="$(arg ransac_n)" />
      <param name="ransac_iter" value="$(arg ransac_iter)" />
  </node>

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find uoais)/ros_nodes/rs_d435.rviz" />
</launch>
